pkg/controller/compliance/compliance_controller.go

```diff
@@ -184,11 +185,17 @@ func (r *ReconcileCompliance) Reconcile(ctx context.Context, request reconcile.R
 		return reconcile.Result{}, err
 	}
 
-	if err = utils.CheckLicenseKey(ctx, r.client); err != nil {
+	license, err := utils.FetchLicenseKey(ctx, r.client)
+	if err != nil {
 		r.status.SetDegraded("License not found", err.Error())
 		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
 	}
 
+	if !utils.IsFeatureActive(license, common.ComplianceFeature) {
+		log.Info("Compliance is not activated as part of this license")
+		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
```

Comment:
Shouldn't we be calling SetDegraded here? I thought that was the design.

Reply:
```
kubectl get tigerastatus intrusion-detection -o yaml

apiVersion: operator.tigera.io/v1
kind: TigeraStatus
metadata:
  creationTimestamp: "2021-02-10T19:16:11Z"
  generation: 1
  managedFields:
  - apiVersion: operator.tigera.io/v1
    fieldsType: FieldsV1
    fieldsV1:
      f:spec: {}
      f:status:
        .: {}
        f:conditions: {}
    manager: operator
    operation: Update
    time: "2021-02-10T19:16:11Z"
  name: intrusion-detection
  resourceVersion: "196988"
  selfLink: /apis/operator.tigera.io/v1/tigerastatuses/intrusion-detection
  uid: bc96282e-5e05-41e5-9461-7ea2c9ac0615
spec: {}
status:
  conditions:
  - lastTransitionTime: "2021-02-10T19:16:11Z"
    status: "False"
    type: Progressing
  - lastTransitionTime: "2021-02-10T19:16:11Z"
    message: License does not support this feature
    reason: Feature is not active
    status: "True"
    type: Degraded
  - lastTransitionTime: "2021-02-10T19:16:16Z"
    status: "False"
    type: Available

```